hdat/hdat_cli.py

```diff
@@ -113,3 +124,90 @@ def diff_results(suites, golden_result, result):
         traceback.print_exc()
         msg = 'Error when attempting to show "{}": {}'
         raise AbortError(msg.format(print_resultspec(result), e))
+
+
+def get_result_keys(result, input_key_list):
+    keys = []
+
+    for in_key in input_key_list:
+        if ".*" in in_key:
+            nested_in_key = in_key.replace(".*", "")
+            for result_key in result.keys():
+                if nested_in_key in result_key and isinstance(result[result_key], dict):
+                    for nested_key in result[result_key].keys():
+                        keys.append(".".join([nested_in_key, nested_key]))
+        elif in_key in result.keys():
+            keys.append(in_key)
+    return sorted(keys)
+
+
+def get_unused_keys(input_key_list, distinct_keys):
+    unused_keys = list(set(input_key_list)-set(distinct_keys))
+    for undefined_key in unused_keys:
+        if ".*" in undefined_key:
+            base_key, wildcard = undefined_key.split(".")
+            for key in distinct_keys:
+                if base_key in key and "." in key:
+                    unused_keys.remove(undefined_key)
+                    break
+    return sorted(unused_keys)
+
+
+def get_result_data(key, result):
+    if "." in key:
+        base_key, nested_key = key.split(".")
+        if isinstance(result[base_key], dict):
+            if nested_key in result[base_key].keys():
+                return str(result[base_key][nested_key])
+    elif key in result.keys():
+        return str(result[key])
+    else:
+        return False
+
+
+def build_result_dict(distinct_keys, results_list):
```

Comment:
I think this should be named something like `build_result_csv_dict` or `build_result_output_dict`, since it's tailored specifically for that use (you probably wouldn't want the blankspace keys if you're doing further processing with this dict)